scionlab/tests/test_user_as_models.py

```diff
@@ -734,15 +910,23 @@ def setUp(self):
 
     def test_delete_single(self):
         seed = 456
-        user_as = create_random_useras(self, seed=seed)
+        r = random.Random(seed)
+        vpn_choice = VPNChoice.SOME
+        all_defs = get_all_random_as_defs()
+        attachment_points = as_defs2aps(all_defs[r.randint(0, len(all_defs)-1)])
+        aps_confs = _get_random_aps_params(seed, attachment_points, vpn_choice)
+        user_as = create_and_check_random_useras(self,
+                                                 seed,
+                                                 aps_confs=aps_confs,
+                                                 vpn_choice=vpn_choice,
+                                                 owner=get_testuser())
         user_as_hosts = list(user_as.hosts.all())
-        attachment_point = user_as.attachment_point
-
         user_as.delete()
 
         self.assertEqual(
             list(Host.objects.needs_config_deployment()),
-            sorted(user_as_hosts + list(attachment_point.AS.hosts.all()),
+            sorted(user_as_hosts +
+                   list(set(h for c in aps_confs for h in c.attachment_point.AS.hosts.all())),
```

Comment:
Why not use the `flatten` here?

Reply:
Since for each `c` in `aps_confs` you get all the `c.attachment_point`s, you don't directly loop over `c`, so flatten won't work